crates/router/src/core/payment_methods/cards.rs

```diff
@@ -333,18 +335,24 @@ pub async fn delete_card<'a>(
     let request = payment_methods::mk_delete_card_request(&state.conf.locker, merchant_id, card_id)
         .change_context(errors::ApiErrorResponse::InternalServerError)
         .attach_printable("Making Delete card request Failed")?;
+
+    let card_delete_failure = "Failed while deleting card from card_locker";
     let delete_card_resp = if !locker.mock_locker {
         services::call_connector_api(state, request)
             .await
-            .change_context(errors::ApiErrorResponse::InternalServerError)?
-            .map_err(|_x| errors::ApiErrorResponse::InternalServerError)?
+            .change_context(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable(card_delete_failure)?
+            .map_err(|_x| report!(errors::ApiErrorResponse::InternalServerError))
+            .attach_printable(card_delete_failure)?
```

Comment:
At this point we're losing information of the previous error. Use a `change_context()` itself, instead of `map_err()`.